x-pack/plugins/security_solution/public/common/components/endpoint/route_capture.tsx

```diff
@@ -18,6 +20,10 @@ export const RouteCapture = memo(({ children }) => {
   const location: AppLocation = useLocation();
   const dispatch = useDispatch();
 
+  useEffect(() => {
```

Comment:
This change is a revert

Reply:
this dispatch looks like it should be somewhere inside a middleware and not on this component. This component can easily become cluttered with unrelated domain-specific logic if we were to follow this pattern.

is there a timeline id middleware where you can instead listen to the userChangedUrl and dispatch this action there?

Reply:
I agree with you. This changed was reintroduced because I reverted a fix that had moved it to somewhere else. Today most routing logic of security threat hunting is inside `x-pack/plugins/security_solution/public/common/components/url_state/use_url_state.tsx`. I am working on refactoring this file, so I will consider moving this change back there.